agency_swarm/agency/agency.py

```diff
@@ -349,25 +423,62 @@ def handle_file_upload(file_list):
                                     "file_id": file.id,
                                     "tools": get_tools(file.filename)
                                 })
-
-                            message_file_names.append(file.filename)
-                            print(f"Uploaded file ID: {file.id}")
+                                
+                            if file_path.endswith('.zip'):
+                                extracted_files.extend(extract_zip(file_path, to_folder))
```

Comment:
Thank you for your contribution, @sjurajpuchky ! Could you explain why you consider this functionality part of the core agent orchestration framework? Could it be implemented separately, either as a tool or a standalone service?

Reply:
Hi bonk1t, like to happen. Main reason was to availability upload more files at once with zip file since, there is a limit of 10 files (what i would like to solve in future) and one zip file is more confortable.

In date of PR, was zip files not supported.

Reply:
@sjurajpuchky Could you update the PR to include only the minimal necessary changes? I’ll review and merge it. Cheers!